tools/nxfuse/mksmartfsimg.sh

```diff
@@ -67,22 +67,32 @@ mkdir $CONTENTSDIR/$BOARDNAME/mnt/bins
 fi
 fi
 
+echo "make a dummy"
 #Make a dummy .bin file
 touch $BINDIR/$BINNAME
 
+echo "make image file"
 # Making image file
-dd if=/dev/zero of=$BINDIR/$BINNAME bs=$blksize count=$blkcount
+if [ "${CONFIG_SMARTFS_ERASEDSTATE}" == 0xff ]; then
```

Comment:
Why now is there a need to translate "\000" to "\377" while creating unformatted image?
Currently it works fine too I believe...

Or have you added this to create correct unformatted image in the 1st time itself? (With erasedstate = 0xFF)

Reply:
That means set 8bit to 0xff, 377 is 0xff for octal.

Reply:
Adding the condition to check erasedstate = 0xff
It will create unformatted image with all bytes set to 0xff initially instead of 0x00, right?

That is the intention with this change

Reply:
So with this, we can actually save one extra step of first erasing the entire partition to 0xff
If it is initially formatted to 0x00

Reply:
yes to prevent an error when write journal data at the first time, as I told you flow of nxfuse is not aligned with smartfs's flow.
So it try to record first journal data when it llformat.